product_pack/models/product_product.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+import math
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    pack_line_ids = fields.One2many(
+        'product.pack.line',
+        'parent_product_id',
+        'Pack Products',
+        help='List of products that are part of this pack.'
+    )
+    used_pack_line_ids = fields.One2many(
+        'product.pack.line',
+        'product_id',
+        'On Packs',
+        help='List of packs where product is used.'
+    )
+
+    @api.multi
+    def separate_pack_products(self):
+        """ Divide the products and the pack products into two separate
+        recordsets.
+        :return: [packs, no_packs]
+        """
+        packs = self.filtered(lambda p: p.pack and p.pack_type in [
```

Comment:
Always use `set`s when using `in`. It's a lot faster.
```suggestion
        packs = self.filtered(lambda p: p.pack and p.pack_type in {
```